src/aggregator/client/tcp_client.go

```diff
@@ -301,7 +302,9 @@ func (c *TCPClient) write(
 			multiErr = multiErr.Add(err)
 			continue
 		}
-
+		// using unsfae.Sizeof to avoid the overhead of marshaling the payload to bytes
+		// tradeoff is that this is not 100% accurate, but should be close enough
+		c.metrics.totalBytesSent.Inc(int64(unsafe.Sizeof(payload)))
```

Comment:
using `unsafe.Sizeof` is going to be way off. 

> Sizeof takes an expression x of any type and returns the size in bytes of a hypothetical variable v as if v was declared via var v = x. **The size does not include any memory possibly referenced by x.** For instance, if x is a slice, Sizeof returns the size of the slice descriptor, not the size of the memory referenced by the slice; if x is an interface, Sizeof returns the size of the interface value itself, not the size of the value stored in the interface. For a struct, the size includes any padding introduced by field alignment. The return value of Sizeof is a Go constant if the type of the argument x does not have variable size. (A type has variable size if it is a type parameter or if it is an array or struct type with elements of variable size).

via [godocs](https://pkg.go.dev/unsafe#Sizeof)

Reply:
as part of writing it out (the `.Write()` call above), the code is already traversing the object to encode it 

it'll be a minor change to bubble up the difference in encoder length, after writing from this [method](https://github.com/m3db/m3/blob/master/src/aggregator/client/writer.go#L170). 

Please do that instead.

Reply:
Ah great point. I will update